tests/test_chain.py

```diff
@@ -303,3 +304,56 @@ def test_deposit_replacement(lianad, bitcoind):
     addr = lianad.rpc.getnewaddress()["address"]
     bitcoind.rpc.sendtoaddress(addr, 2)
     wait_for(lambda: len(lianad.rpc.listcoins()["coins"]) == 1)
+
+def test_rescan_and_recovery(lianad, bitcoind):
+    """Test user recovery flow"""
+    # Get initial_tip to use for rescan later
+    initial_tip = bitcoind.rpc.getblockheader(bitcoind.rpc.getbestblockhash())
+
+    # Start by getting a few coins
+    destinations = {
+        lianad.rpc.getnewaddress()["address"]: 0.5,
+    }
+    txid = bitcoind.rpc.sendmany("", destinations)
+    bitcoind.generate_block(1, wait_for_mempool=txid)
+    wait_for(
+        lambda: lianad.rpc.getinfo()["block_height"] == bitcoind.rpc.getblockcount()
+    )
+    assert len(lianad.rpc.listcoins()["coins"]) == 1
+
+    # Advance the blocktime by >2h in median-time past for rescan
+    added_time = 60 * 60 * 3
+    bitcoind.rpc.setmocktime(initial_tip["time"] + added_time)
+    bitcoind.generate_block(12)
+
+    # Clear lianad state
+    lianad.restart_fresh(bitcoind)
+    assert len(lianad.rpc.listcoins()["coins"]) == 0
+
+    # Start rescan
+    lianad.rpc.startrescan(initial_tip["time"])
+    wait_for(lambda: lianad.rpc.getinfo()["rescan_progress"] is None)
+
+    # Receive another coin, it will be Ten blocks later
+    txid = bitcoind.rpc.sendtoaddress(lianad.rpc.getnewaddress()["address"], 0.4)
+    bitcoind.generate_block(1, wait_for_mempool=txid)
```

Comment:
What's the reason for receiving another coin here? Since this is testing the typical recovery flow i'd expect the user would receive multiple coins at different heights *before* losing the state (i.e. at the beginning of the test), not after recovering his wallet.

Reply:
I agree. This part doesn't do anything useful.